src/uu/stat/src/stat.rs

```diff
@@ -110,6 +47,81 @@ pub const F_SIGN: u8 = 1 << 4;
 // unused at present
 pub const F_GROUP: u8 = 1 << 5;
 
+/// checks if the string is within the specified bound
+///
+fn check_bound(slice: &str, bound: usize, beg: usize, end: usize) -> UResult<()> {
+    if end >= bound {
+        return Err(USimpleError::new(
+            1,
+            format!("{}: invalid directive", slice[beg..end].quote()),
```

Comment:
I'm not too sure, what this function does except, error-ing out if we try to index a string greater then the `bound`(which is the size of the string),  in which case, shouldn't `slice[beg..end]` have been `slice[beg..bound]` instead, since, `end` could be arbitrarily any value in this context(in which case we are accessing an invalid index). also, apologies if this sounds stupid :P , just couldn't wrap my head around it? would be very glad to get some input/suggestion on this. Thanks! @tertsdiepraam @sylvestre